shared/AppInsightsCommon/src/Offline.ts

```diff
@@ -98,9 +114,20 @@ export function createOfflineListener(parentEvtNamespace?: string | string[]): I
         }
     }
 
+    function addListener(callback: OfflineCallback) {
+        listenerList.add(callback);
+        return {
+            rm: () => {
```

Comment:
As above, the list should be an array and an edge case that has to be handled is that someone "could" call this `rm` function multiple times, so the first time will remove the entry (if present) and subsequent times will do nothing.

On way to do this is to use a temporary variable and then when `rm` is called you just reassign the `rm` function on that variable (that you return) to a function that does nothing.

Reply:
function addListener(callback: OfflineCallback) {
        listenerList.push(callback);
        const rm = (callback: OfflineCallback) => {
            let removed = false;
            return () => {
                if (!removed) {
                    const index = listenerList.indexOf(callback);
                    if (index !== -1) {
                        // Remove the callback from the list
                        listenerList.splice(index, 1);
                        removed = true; // Mark it as removed
                    }
                }
            };
        };
        return rm;
   does this works?

Reply:
seems like it does not work, vs is complaining the type error